src/main/groovy/org/pih/warehouse/api/StockMovement.groovy

```diff
@@ -179,6 +184,7 @@ class StockMovement implements Validateable{
             trackingNumber      : trackingNumber,
             driverName          : driverName,
             comments            : comments,
+            recentComment       : recentComment,
```

Comment:
I agree with you something doesn't feel right about that - I mean, we only care about the `shipmentComments` - what if we decided to care about the requisition comments also?
Would we then want to distinguish shipment/requisition recent comment or how would we want to approach that?
I don't know the scope much to be able to answer that question.

Reply:
yes, the whole comments hierarchy should be reevaluated like, which comments does the `StockMovement` inherit? The Shipment comments? Requisition comments? Order Comments?
Maybe we should mash all of those comments in a common list and extract the most recent one regardless of it's origin?

Reply:
Yeah, we need to rewrite the APIs completely. This is a case where we've overloaded StockMovement to do so many things and it's showing signs of collapse.

Reply:
Can you explain what "recentComment" is for? I know it's probably related to the requisition approval, but I want to know specific use case without digging through the code. I suspect this is a case where the codes needs to be more specific about what we're trying to accomplish i.e. getApprovalComment. And that makes me feel like we need to do something similar to synonyms where we need to classify comments according to their purpose instead of just getting the latest comment.